### DISCUSSION TITLE
pac autocomplete

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/powerplatform-vscode#567] [Ideas]
I see that documentation has been added about pac autocomplete https://learn.microsoft.com/en-us/power-platform/developer/cli/reference/#enable-tab-completion-in-powershell

pac native autocomplete when I press pac Ctrl+Space
![image](https://github.com/microsoft/powerplatform-vscode/assets/5035266/66bfb4a6-e2c9-4306-9c76-4778db41d0fc)

my autocomplete when I press pac Ctrl+Space
![image](https://github.com/microsoft/powerplatform-vscode/assets/5035266/f90d60fe-6b8f-47db-841b-0437155556ec)

I am not getting a list of suggestions like the custom one I did (https://twitter.com/RajYRaman/status/1604640181885468672?s=20). Can this please be added to the auto-complete capability? It would be hard to keep mine up to date with all the changes in pac, so please add this to native auto-complete.

**Comment by petrochuk** [comment]
Latest [Microsoft.PowerApps.CLI](https://nuget.info/packages/Microsoft.PowerApps.CLI/1.24.3) packages include pac.doc.json which contains auto generated list of all supported commands and arguments.

**Reply by rajyraman** [reply]
@petrochuk - Thank you for pointing out this file. I can use this as the source for my PS module and slighly tweak it. However, native autocomplete does not seem to show the list of suggestions like I posted in the screenshot even with the json file. Should I raise this as an issue, or is this feature not officially released yet as there is no mention of it in @kkanakas [latest post](https://devblogs.microsoft.com/powerplatform/power-platform-developer-tools-may-update/)?

**Reply by petrochuk** [reply]
We documented a way to hook up with auto complete functionality different shells provide. It is similar to [dotnet complete](https://learn.microsoft.com/en-us/dotnet/core/tools/enable-tab-autocomplete) when you can enable auto complete in bash, zsh, fish, and nushell.

The best way to ask for changes is to post an idea in discussions section and get votes for it. We are regularly looking at this list.

**Reply by rajyraman** [reply]
@petrochuk - That was quick to update. 10 secs to find and replace and released a new version today. Thank you for pointing out this file Andrew. I still like mine with PSReadline suggestions 😄.
